content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,88 @@
+
+IPFS first separates your content into _blocks_. If you had a huge file with a single CID, you'd have to transfer the whole CID every time someone requested it. But Instead, you can break it into blocks, and transfer it block by block instead of one massive chunk at a time. Each block might come from a different source, even! This is a much more efficient way to transmit data. (If you've used BitTorrent, you may have noticed that when you download a file, it fetches it from a bunch of random people; this is the same idea.)
+
+Another great feature of the DAG and breaking content into blocks is that if you have two similar files, they can share the DAG; ie, different trees can reference the same data. For example, if you copyedit a document and change a few commas here and there, IPFS knows to only create new blocks for those small changes. Your old version and your new version can refer to the same blocks for everything else. This can make transferring versions of large datasets (such as genomics research or weather data) much more efficient because you only need to transfer the parts that are new or have changed instead of creating entirely new files each time.
+
+DAGs are a bit of a ["turtles all the way down"](https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/Turtles_all_the_way_down.html) scenario; that is, **everything** has a CID! You've got a file that has a CID. What if there are several files in a folder? That folder has a CID, and that CID contains the CIDs of the files underneath. In turn, those files are made up of blocks, and each of those blocks has a CID. You can see how a file system on your computer could just be a DAG. You can also, hopefully, how DAG trees start to form.
+
+Another important aspect of DAGs to note: if you change a small part of a DAG, the whole root will become different. Take a look at [this helpful illustration using bananas](https://media.consensys.net/ever-wonder-how-merkle-trees-work-c2f8b7100ed3) from our friends at Consensys.
+
+### 3 \\ The DHT
+
+<img alt="IPFS Stack - Routing and Exchange" src="../assets/ipfs_stack-exchange_routing.png" width="200px" />
+
+So, to recap, IPFS lets you to give CIDs to content, and link that content together by generating DAGs using IPLD. Now let's move on to the last piece: how you find and move that content around!
+
+In order to find which peer has certain content (_discovery_), you use a [_distributed hash table_](https://en.wikipedia.org/wiki/Distributed_hash_table), or DHT. **IPFS is essentially a database.** Just like key-value pairs, if you have a peer, it will let you find content, and if you have content, it will let you find you the peer where the content is stored.
```

Comment:
```suggestion
To find which peers hosting content you are after (_discovery_), IPFS uses a [_distributed hash table_](https://en.wikipedia.org/wiki/Distributed_hash_table), or DHT. A hash-table is a database of keys to values. A distributed one is where the table is split up across all the peers. To find out where some content is you ask you peers.
```

Reply:
maybe that's not quite the sentiment you wanted here, but i think its important that we dont say _IPFS is a database_. Also it is not common in comuter world that you map both ways... keys typically get you to values but values rarely link back to keys.

Reply:
Tell me more 😄 Why do you think it's important that we don't say _IPFS is a database_ @olizilla ?

`Also it is not common in comuter world that you map both ways... keys typically get you to values but values rarely link back to keys.` Noted. Reworking, also based on Hector's feedback.

Reply:
This was reworked with this suggestion incorporated.